project-koku/koku
# COST-1639: Introducing report limit=0 query option to return all data.

**Testing**
1. Run smokes
2. Execute an CSV export via the API with Content-Type: text/csv to generate the CSV file.  Do a query with no parameter (default 100 limit) and another query with limit=0 to verify the full response is returned.

**Test Results**
1. Smoke Results
```
================================================================================================================== short test summary info ===================================================================================================================
FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test__ocp_cost_reports.py::test_api_ocp_cost_order_by_name_pagination - AssertionError: Failing from https://issues.redhat.com/browse/COST-1734
FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test_azure_cost_reports.py::test_api_azure_cost_tag_validate_keys_visible - AssertionError: Failing from https://issues.redhat.com/browse/COST-1452 if still open
FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test_gcp_cost_reports.py::test_api_gcp_cost_order_by_name_pagination - AssertionError: Failing from https://issues.redhat.com/browse/COST-1734
========================================================================================== 3 failed, 5831 passed, 201 skipped, 648 deselected in 3354.03s (0:55:54) ==========================================================================================
```

2. CSV export with no limit set (100 default)
<img width="1268" alt="image" src="https://user-images.githubusercontent.com/4523309/128765247-97f7153b-7ffa-417a-88e4-672c5ce2638c.png">


CSV export with limit = 0

<img width="1268" alt="image" src="https://user-images.githubusercontent.com/4523309/128765318-b78eb199-3796-4b69-b9c9-a15f28d8532e.png">
